e2e-tests/spring-rest-mongo/src/test/java/org/evomaster/e2etests/spring/rest/mongo/document/BsonDocumentWithMREMTest.java

```diff
@@ -0,0 +1,50 @@
+package org.evomaster.e2etests.spring.rest.mongo.document;
+
+import com.foo.spring.rest.mongo.BsonDocumentAppController;
+import org.evomaster.core.EMConfig;
+import org.evomaster.core.problem.rest.HttpVerb;
+import org.evomaster.core.problem.rest.RestIndividual;
+import org.evomaster.core.search.Solution;
+import org.evomaster.e2etests.utils.RestTestBase;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+
+public class BsonDocumentWithMREMTest extends RestTestBase {
+
+    @BeforeAll
+    public static void initClass() throws Exception {
+        EMConfig config = new EMConfig();
+        config.setInstrumentMR_MONGO(true);
+        RestTestBase.initClass(new BsonDocumentAppController(), config);
+    }
+
+    @Test
+    public void testRunEMWithMongoMethodReplacement() throws Throwable {
+        runTestHandlingFlakyAndCompilation(
+                "BsonDocumentWithMREM",
+                "org.foo.spring.rest.mongo.BsonDocumentEMGeneration",
+                1000,
+                (args) -> {
+                    args.add("--enableWeightBasedMutationRateSelectionForGene");
+                    args.add("false");
+                    args.add("--heuristicsForMongo");
+                    args.add("false");
+                    args.add("--instrumentMR_MONGO");
+                    args.add("true");
+                    args.add("--generateMongoData");
+                    args.add("false");
+                    args.add("--extractMongoExecutionInfo");
+                    args.add("false");
```

Comment:
new E2E tests should rather use the new `setOption` method introduced in `EnterpriseTestBase` (see for example `BBAuthEMTest`)

Reply:
fixed